### DISCUSSION TITLE
New NoteBook Property = Default Wiki Template

### DISCUSSION THREAD

**Discussion by nh3oh** [zim-desktop-wiki/zim-desktop-wiki#2701] [Ideas]
Hi!

_This may be yet somewhere else in the pipework but I didn't find it._

The concept is mostly exposed in the title.

Should be work and time saving while creating new pages in a determinate notebook,
not to have to scroll the template list,
and even greater when generating new pages right from links while editing,
**that will open/be created with the dedicated template** and not the actual "standard" default one,
avoiding the hassle of trashing an empty new page only to recreate it with the right template.

Best regards.

**Comment by jaap-karssenberg** [comment]
The idea is that you can edit the default template if you want to change
the default page layout

Op za 30 nov. 2024 16:39 schreef nh3oh ***@***.***>:

**Comment by nh3oh** [comment]
_If I do not misunderstand you, the idea should be to edit the default template file each time I change of current NoteBook (NB) (10x per day) ???
... hoping I misunderstand you._

The problem I encounter is exactly that the default wiki template is actually the default for every NB created in my user space.

As I use NBs for various purposes, one only default template do not cover different contexts ;
knowing that each context usually requires one "regular" template used for most of those BN pages.

In the past -- I still keep some notes about -- the Profile BN property was probably able to point toward an alternative default template, but profiles seem not to be supported any more.

Perhaps they were not enough desktop's user friendly?... so think I.

Then, building-up a new property easy to set, may be worthy.

**Comment by nh3oh** [comment]
Practically could be :

Adding a new NB to local Zim `notebooks.list` creates a copy of current `default.txt` wiki template named (p.ex.) `<interwiki>.txt` that will be used as default for the `Default Wiki Template` NB property.

This way, one can edit this new template as required and starts to use the NB without thinking more about it.
